### ISSUE TITLE
Filesystem Free Space check doesn't take into account post-download?

### ISSUE THREAD

**Issue by KWall24** [grahampugh/erase-install#242]
**Describe the bug**
This script works wonderfully, save for if the device has ~40GB or so of space - just enough to download the installer, but not enough to run the installer. When it does this, it only checks free space once at the beginning of the script prior to the download (unless I'm missing something). It seems to need to check after as well. I have experienced this myself, as well as a few users that I'm trying to deploy it to. I'm including the log below.

**To Reproduce**
 Running script from Jamf Policy - upload erase-install-26.0.pkg and use Files and Processes to run it with `/Library/Management/erase-install/erase-install.sh --os=12 --update --reinstall --confirm --check-power`
 
**Expected behavior**
Script checks for free space, downloads install file, installs it with no hiccups
Or script checks for free space, finds not enough, warns user, exits.

**Actual behavior**
Script checks free space, finds enough space for download, successfully downloads, finds not enough space for install, no warning and stays stuck on full-screen "Preparing your mac. This could take 30 minutes" jamfHelper screen indefinitely until forced reboot.

[erase-install.log](https://github.com/grahampugh/erase-install/files/8689038/erase-install.log)

**Environment (please complete the following information):**
 - OS version High Sierra thru Big Sur.

 - erase-install version v26.0

**Additional context**
If I'm missing something, please let me know. I wouldn't be surprised that this is working as intended and I'm just dumb. Otherwise I love this script, it is actively making my job easier(even with this issue in play)

**Comment by itsbradwhite** [comment]
Just wanted to say, we've been seeing this too. We just started using this script not long ago. First few computers I tested it on went without a hitch. But after that, the next few all got disk space errors. One computer had 32 gigs of free space.

Computer with 32 gigs of free space:

```
[STEP 4 of 5]
Running command /Library/Management/erase-install/erase-install.sh --update --reinstall --overwrite --current-user --depnotify --cleanup-after-use...
Result of command:[erase-install] v26.0 script execution started: Tue May 17 14:48:57 PDT 2022    [erase-install] Caffeinating this script (pid=4836)    [check_free_space] ERROR - 34 GB free/purgeable disk space detected 0    [erase-install] attempting to terminate the 'caffeinate' process - Termination message indicates success /Library/Management/erase-install/erase-install.sh: line 1042:  4849 Terminated: 15          /usr/bin/caffeinate -dimsu -w $
```
Above was a coworkers M1 laptop already running Monterey 12.3.1, he was just testing the script. Thought for sure he'd have enough free space, but still threw that error. This error happened before the download started.

This user had like 60+ gigs of free space and it still said it needed an additional 400Mb. This user was already on Big Sur, don't recall what version, just trying to update to Monterey and had the volume permission pop-up, so this script was ran to get around that. Her ouput (happened after Monterey installer was downloaded):

```
Product downloaded and installed to /Library/Management/erase-install/Install_macOS_12.3.1-21E258.sparseimage
   [run_installinstallmacos] Mounting sparse disk image to identify installer app.
/dev/disk5          	GUID_partition_scheme          	
/dev/disk5s1        	EFI                            	
/dev/disk5s2        	Apple_HFS                      	/Volumes/Install_macOS_12.3.1-21E258
   [erase-install] Installer is at: /Volumes/Install_macOS_12.3.1-21E258/Applications/Install macOS Monterey.app
   [erase-install] Closing DEPNotify download message (language=en)
/Library/Management/erase-install/erase-install.sh: line 789: kill: (12605) - No such process

   [erase-install] WARNING! Running /Volumes/Install_macOS_12.3.1-21E258/Applications/Install macOS Monterey.app with reinstall option

   [erase-install] Opening DEPNotify message (language=en)
   [erase-install] Writing LaunchDaemon which will remove /Library/Management/erase-install at next boot
Error: The target volume does not have enough free space to install.
An additional 439.9 MB of space is needed to install on this disk.
By using the agreetolicense option, you are agreeing that you have run this tool with the license only option and have read and agreed to the terms.
If you do not agree, press CTRL-C and cancel this process immediately.

   [erase-install] attempting to terminate the 'caffeinate' process - Termination message indicates success
/Library/Management/erase-install/erase-install.sh: line 1042: 12493 Terminated: 15          /usr/bin/caffeinate -dimsu -w $$
```

Another user with Big Sur (11.3.1) trying to update to Monterey with ~25Gb of free space. (Error happened before Monterey was downloaded):

```
[STEP 4 of 5]
--
Running command /Library/Management/erase-install/erase-install.sh --update --reinstall --overwrite --current-user --depnotify --cleanup-after-use...
Result of command:[erase-install] v26.0 script execution started: Tue May 17 13:56:05 PDT 2022    [erase-install] Caffeinating this script (pid=19363)    [check_free_space] ERROR - 25 GB free/purgeable disk space detected 0    [erase-install] attempting to terminate the 'caffeinate' process - Termination message indicates success /Library/Management/erase-install/erase-install.sh: line 1042: 19376 Terminated: 15          /usr/bin/caffeinate -dimsu -w $
```

So that's two users with over 25Gb of free space, but script if failing to download Monterey installer. Error happens on the initial check_free_space. Is this similar to what you're getting?

TL:DR - Script ran flawlessly on multiple computers, I thought it was good, next few get disk space errors when it seems like surely they have enough.

**Comment by grahampugh** [comment]
Please try the latest 26.1 upload.

**Comment by grahampugh** [comment]
Is that running at the login window?

**Comment by grahampugh** [comment]
The `0` at the end of the line is peculiar, but you need 45 GB of space by default. If you want to try reducing it, you can override with `--min-drive-space`.

**Comment by itsbradwhite** [comment]
How would you suggest using the --min-drive-space flag? I didn't realize it needed a parameter, so it probably hasn't been doing anything. The most Monterey can possibly need is 45GB I believe. What would happen if you did 
```
--min-drive-space 40
```

**Comment by grahampugh** [comment]
If you don't give `--min-drive-space` a value, whatever is next in the command would be the value, so you may have been getting unintended problems. e.g. if you said `--min-drive-space --current-user`, then `--min-drive-space` won't work and `--current-user` will be ignored.

It's up to you to test other values. The default is set to 45, so if you're happy with that value, leave out the parameter completely. If you want to try and squeeze out a smaller space, go ahead. I never use this parameter.

**Comment by KWall24** [comment]
That isn't quite what I'm seeing, but that sucks too.

What I'm seeing can be boiled down to failing after the download (after the
actual Mac installer starts it get upset and can't continue cause there
isn't enough space). Sounds like yours fails prior to the download.
